seam/client.py

```diff
@@ -30,10 +34,34 @@ def request(self, method, url, *args, **kwargs):
         return self._handle_response(response)
 
     def _handle_response(self, response: requests.Response):
-        if response.status_code != 200:
-            raise SeamHttpApiError(response)
+        if not 200 <= response.status_code < 300:
+            self._handle_error_response(response)
 
-        if "application/json" in response.headers["content-type"]:
+        if "application/json" in response.headers.get("content-type", ""):
             return response.json()
 
         return response.text
+
+    def _handle_error_response(self, response: requests.Response):
+        status_code = response.status_code
+        request_id = response.headers.get("seam-request-id")
+
+        if status_code == 401:
+            raise SeamHttpUnauthorizedError(request_id)
+
+        if status_code != 200:
+            error = response.json().get("error", {})
+            error_type = error.get("type", "unknown_error")
+            error_message = error.get("message", "Unknown error")
+            error_data = error.get("data", None)
+
+            error_details = {
+                "type": error_type,
+                "message": error_message,
+                "data": error_data,
+            }
+
+            if error_type == "invalid_input":
+                raise SeamHttpInvalidInputError(error_details, status_code, request_id)
+
+            raise SeamHttpApiError(error_details, status_code, request_id)
```

Comment:
The `!200` check is duplicated as the above only calls this method on non-200s.

```suggestion
        error = response.json().get("error", {})
        error_type = error.get("type", "unknown_error")
        error_message = error.get("message", "Unknown error")
        error_data = error.get("data", None)

        error_details = {
            "type": error_type,
            "message": error_message,
            "data": error_data,
        }

        if error_type == "invalid_input":
            raise SeamHttpInvalidInputError(error_details, status_code, request_id)

        raise SeamHttpApiError(error_details, status_code, request_id)
```

Reply:
Whoops, nice catch! Forgot to remove it after a refactor. Thanks!

Reply:
e5c0921851aa9c10dd772d12d115c089bb214a98